core/core.go

```diff
@@ -321,9 +321,8 @@ func setupDiscoveryOption(d config.Discovery) DiscoveryOption {
 
 func (n *IpfsNode) HandlePeerFound(p peer.PeerInfo) {
 	log.Warning("trying peer info: ", p)
-	ctx, _ := context.WithTimeout(context.TODO(), time.Second*10)
-	err := n.PeerHost.Connect(ctx, p)
-	if err != nil {
+	ctx, _ := context.WithTimeout(n.Context(), time.Second*10)
```

Comment:
maybe timeout should be `30s` -- high latency links may take much longer to handshake.